hadml/datamodules/gan_datamodule.py

```diff
@@ -251,3 +249,167 @@ def test_dataloader(self):
             ),
         }
         return CombinedLoader(loaders, mode="max_size_cycle")
+
+
+class MultiHadronEventGANDataModule(LightningDataModule):
+    def __init__(
+            self, 
+            data_dir="data/Herwig",
+            raw_file_list=["AllClusters_10K.dat"],
+            processed_filename="herwig_multihadron_events_10K.npy",
+            pid_map_file="pid_to_idx.pkl",
+            train_val_test_split: Tuple[float, float, float] = (0.7, 0.15, 0.15),
+            batch_size=32,
+            num_workers=8,
+            initialise_data_preparation=False,
+            debug=True
+        ):
+        super().__init__()
+        self.processed_filename = os.path.join(os.path.normpath(data_dir),
+                                               "processed", processed_filename)
+        self.train_val_test_split = train_val_test_split
+        self.batch_size = batch_size
+        self.num_workers = num_workers
+        self.data_train: Optional[torch.utils.data.Dataset] = None
+        self.data_val: Optional[torch.utils.data.Dataset] = None
+        self.data_test: Optional[torch.utils.data.Dataset] = None
+
+        parser = HerwigMultiHadronEventParser(
+            data_dir=data_dir,
+            raw_file_list=raw_file_list,
+            processed_filename=processed_filename,
+            pid_map_file=pid_map_file,
+            debug=debug
+        )
+        parser.parse_data()
+
+        print('\n', ' '*20, self.__class__.__name__,'\n', '-'*70, sep='')
+
+        if initialise_data_preparation:
```

Comment:
we could rename this to something more descriptive, e.g. `log_batch_info`

Reply:
The "initialise_data_preparation" flag makes an instance call its self.prepared_data() and self.setup() methods even when a MultiHadronEventGANDataModule object is not subsequently passed to any model class. Information about the batch size, as well as the first sample sentences, is printed but it is a collateral action. Therefore renaming it may shift our collaborators' attention towards an action which in fact plays an auxiliary role.

Reply:
Ok, thanks for explaining